server/manager/src/test/java/org/apache/accumulo/manager/tableOps/merge/MergeTabletsTest.java

```diff
@@ -0,0 +1,456 @@
+      if (i == 1) {
+        badColumnSetter.accept(builder2);
+      }
+      var tablet2 = builder2.build(LOCATION, LOGS, FILES, ECOMP, MERGED, COMPACTED, SELECTED,
+          USER_COMPACTION_REQUESTED, LOADED, CLONED);
+      var builder3 = TabletMetadata.builder(ke3).putOperation(opid).putDirName("td1")
+          .putTime(MetadataTime.parse("L50")).putTabletAvailability(TabletAvailability.HOSTED);
+      if (i == 2) {
+        badColumnSetter.accept(builder3);
+      }
+      var tablet3 = builder3.build(LOCATION, LOGS, FILES, ECOMP, MERGED, COMPACTED, SELECTED,
+          USER_COMPACTION_REQUESTED, LOADED, CLONED);
+
+      for (var expectedMessage : expectedMessages) {
+        var exception = assertThrows(IllegalStateException.class,
+            () -> testMerge(List.of(tablet1, tablet2, tablet3), tableId, null, null,
+                tabletMutator -> {}));
+        assertTrue(exception.getMessage().contains(expectedMessage),
+            () -> exception.getMessage() + " did not contain " + expectedMessage);
+      }
+    }
+  }
+
+  @Test
+  public void testTime() throws Exception {
+
+    // this test ensures the max time is taken from all tablets
+    String testTimes[][] = {{"L30", "L20", "L10"}, {"L20", "L30", "L10"}, {"L10", "L20", "L30"}};
+
+    for (String[] times : testTimes) {
+
+      var tablet1 = TabletMetadata.builder(ke1).putOperation(opid).putDirName("td1")
+          .putTime(MetadataTime.parse(times[0])).putTabletAvailability(TabletAvailability.HOSTED)
+          .build(LOCATION, LOGS, FILES, ECOMP, MERGED, COMPACTED, SELECTED,
+              USER_COMPACTION_REQUESTED, LOADED, CLONED, SCANS, HOSTING_REQUESTED, SUSPEND, LAST);
+      var tablet2 = TabletMetadata.builder(ke2).putOperation(opid).putDirName("td2")
+          .putTime(MetadataTime.parse(times[1])).putTabletAvailability(TabletAvailability.HOSTED)
+          .build(LOCATION, LOGS, FILES, ECOMP, MERGED, COMPACTED, SELECTED,
+              USER_COMPACTION_REQUESTED, LOADED, CLONED, SCANS, HOSTING_REQUESTED, SUSPEND, LAST);
+      var tablet3 = TabletMetadata.builder(ke3).putOperation(opid).putDirName("td3")
+          .putTime(MetadataTime.parse(times[2])).putTabletAvailability(TabletAvailability.HOSTED)
+          .build(LOCATION, LOGS, FILES, ECOMP, MERGED, COMPACTED, SELECTED,
+              USER_COMPACTION_REQUESTED, LOADED, CLONED, SCANS, HOSTING_REQUESTED, SUSPEND, LAST);
+
+      testMerge(List.of(tablet1, tablet2, tablet3), tableId, null, null, tabletMutator -> {
+        EasyMock.expect(tabletMutator.putTime(MetadataTime.parse("L30"))).andReturn(tabletMutator)
+            .once();
+        EasyMock.expect(tabletMutator.putTabletAvailability(TabletAvailability.HOSTED))
+            .andReturn(tabletMutator).once();
+        EasyMock.expect(tabletMutator.putPrevEndRow(ke1.prevEndRow())).andReturn(tabletMutator)
+            .once();
+        EasyMock.expect(tabletMutator.setMerged()).andReturn(tabletMutator).once();
+      });
```

Comment:
Should there be a EasyMock.verify(...) in here?

Reply:
There is a call to verify outside of the lambda.